applications/FreeSurfaceApplication/python_scripts/edgebased_levelset_solver.py

```diff
@@ -0,0 +1,350 @@
+        self.use_mass_correction = True
+        self.redistance_frequency = 5
+        self.step = 0
+        self.extrapolation_layers = 5
+        self.tau2_factor = 0.0
+        self.edge_detection_angle = 45.0
+        self.assume_constant_pressure = True
+        self.timer = Timer()
+
+        self.use_parallel_distance_calculation = False
+        # 0 = None; 1 = Ergun; 2 = Custom A y B;
+        self.compute_porous_resistance_law = 0
+
+        # neighbour search
+        number_of_avg_elems = 10
+        number_of_avg_nodes = 10
+        self.neighbour_search = FindNodalNeighboursProcess(
+            model_part, number_of_avg_elems, number_of_avg_nodes)
+        (self.neighbour_search).Execute()
+
+        # erase isolated notes
+        eliminate_isolated = EliminateIsolatedNodesProcess(model_part)
+        eliminate_isolated.Execute()
+
+        # definition of the solvers
+#        pDiagPrecond = DiagonalPreconditioner()
+#        self.pressure_linear_solver =  CGSolver(1e-3, 5000,pDiagPrecond)
+ #       self.pressure_linear_solver =  CGSolver(1e-3, 5000)
+
+#        pDiagPrecond = DiagonalPreconditioner()
+#        self.pressure_linear_solver =  BICGSTABSolver(1e-3, 5000,pDiagPrecond)
+        self.pressure_linear_solver = BICGSTABSolver(1e-3, 5000)
+
+        # initializing the press proj to -body_force
+        press_proj_init = Vector(3)
+        press_proj_init[0] = body_force[0] * density
+        press_proj_init[1] = body_force[1] * density
+        press_proj_init[2] = body_force[2] * density
+        for node in self.model_part.Nodes:
+            eps = node.GetSolutionStepValue(POROSITY)
+            node.SetSolutionStepValue(PRESS_PROJ, 0, press_proj_init * eps)
+        print("entered in EdgeBasedLevelSetSolver initialize")
+
+    def Initialize(self):
+        print("entered in EdgeBasedLevelSetSolver python constructor")
+        # build the edge data structure
+        if(self.domain_size == 2):
+            self.matrix_container = MatrixContainer2D()
+        else:
+            self.matrix_container = MatrixContainer3D()
+        self.matrix_container.ConstructCSRVector(self.model_part)
+        self.matrix_container.BuildCSRData(self.model_part)
+        # for 3D problems we need to evaluate the condition's neighbours
+        if(self.domain_size == 3):
+            self.condition_neighbours_finder = FindConditionsNeighboursProcess(
+                self.model_part, self.domain_size, 10)
+            self.condition_neighbours_finder.Execute()
+        # constructing the solver
+        if(self.domain_size == 2):
+            if(self.use_parallel_distance_calculation == False):
+                self.distance_utils = SignedDistanceCalculationUtils2D()
+            else:
+                self.distance_utils = ParallelDistanceCalculator2D()
+
+            self.fluid_solver = EdgeBasedLevelSet2D(
+                self.matrix_container,
+                self.model_part,
+                self.viscosity,
```

Comment:
Most of the information is on the properties, which is on the modelpart, that way can be encapsulated in fewer arguments